src/Microsoft.Health.Fhir.TemplateManagement.UnitTests/TemplateCollectionProviderFactoryTests.cs

```diff
@@ -75,10 +93,33 @@ public void GivenAWrongDefaultTemplatePath_WhenInitDefaultTemplate_ExceptionWill
         [Fact]
         public void GiveNewDefaultTemplateTarGzFile_WhenInitDefaultTemplate_DefaultTemplatesWillBeInit()
         {
+            var targzName = "NewDefaultTemplates.tar.gz";
             string imageReference = ImageInfo.DefaultTemplateImageReference;
-            CreateTarGz("NewDefaultTemplates.tar.gz", templateFolder);
+            CreateTarGz(targzName, hl7v2TemplateFolder);
+            TemplateCollectionProviderFactory factory = new TemplateCollectionProviderFactory(_cache, Options.Create(_config));
+            factory.InitDefaultTemplates(targzName);
+            Assert.NotNull(factory.CreateProvider(imageReference, string.Empty));
+        }
+
+        [Fact]
+        public void GiveHl7v2NewDefaultTemplateTarGzFile_WhenInitDefaultTemplate_DefaultTemplatesWillBeInit()
+        {
+            var targzName = "Hl7v2NewDefaultTemplates.tar.gz";
+            string imageReference = ImageInfo.GetDefaultTemplateImageReferenceByDatatype(DataType.Hl7v2);
+            CreateTarGz(targzName, hl7v2TemplateFolder);
+            TemplateCollectionProviderFactory factory = new TemplateCollectionProviderFactory(_cache, Options.Create(_config));
+            factory.InitDefaultTemplates(targzName);
+            Assert.NotNull(factory.CreateProvider(imageReference, string.Empty));
+        }
+
+        [Fact]
+        public void GiveCdaNewDefaultTemplateTarGzFile_WhenInitDefaultTemplate_DefaultTemplatesWillBeInit()
```

Comment:
These several `GiveCdaNewDefaultTemplateTarGzFile_..` methods are fairly similar, is it possible to merge them and set the different parts by `MemberData`?

Reply:
done